pkg/controllers/raycluster_controller.go

```diff
@@ -267,6 +274,36 @@ func desiredOAuthSecret(cluster *rayv1.RayCluster, r *RayClusterReconciler) *cor
 	// Create a Kubernetes secret to store the cookie secret
 }
 
+func desiredNetworkPolicy(cluster *rayv1.RayCluster) *networkingapply.NetworkPolicyApplyConfiguration {
+
+	return networkingapply.NetworkPolicy(cluster.Name, cluster.Namespace).
```

Comment:
is it easier to use `"k8s.io/api/networking/v1"` instead of `"k8s.io/client-go/applyconfigurations/networking/v1"`
```
desiredNetworkPolicy := &netv1.NetworkPolicy{
			TypeMeta: metav1.TypeMeta{
				Kind:       "NetworkPolicy",
				APIVersion: "v1",
			},
			ObjectMeta: metav1.ObjectMeta{
				Name:      name,
				Namespace: name,
				Labels:    map[string]string{"ray.io/cluster": cluster.Name, "ray.io/node-type": "head"},
			},
			Spec: netv1.NetworkPolicySpec{
				Ingress: []netv1.NetworkPolicyIngressRule{
					{
						Ports: []netv1.NetworkPolicyPort{
							{
								Port: &intstr.IntOrString{Type: intstr.Int, IntVal: 6379},
							},
							{
								Port: &intstr.IntOrString{Type: intstr.Int, IntVal: 10001},
							},
						},
						From: []netv1.NetworkPolicyPeer{
							{
								NamespaceSelector: &metav1.LabelSelector{
									MatchLabels: map[string]string{
										"kubernetes.io/metadata.name": "value of application namespace",
									},
								},
								PodSelector: &metav1.LabelSelector{
									MatchLabels: map[string]string{
										"app.kubernetes.io/component": "kuberay-operator",
									},
								},
							},
						},
					},
					{
						Ports: []netv1.NetworkPolicyPort{
							{
								Port: &intstr.IntOrString{Type: intstr.Int, IntVal: 8443},
							},
						},
					},
				},
				PolicyTypes: []netv1.PolicyType{
					netv1.PolicyTypeIngress,
				},
			},
		}
```

Reply:
That's needed for SSA, that brings other advantages, like one-shot server-side patching, conflict resolutions with concurrent controllers like KubeRay, etc.

Reply:
thanks for the tips!
i think we will need to move towards this way in operator's  code as well.

Reply:
Right, from my experience, SSA makes things much simpler from a code standpoint and more robust from a runtime one.